balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -548,7 +575,12 @@ func (b *pickfirstBalancer) updateSubConnState(sd *scData, newState balancer.Sub
 		return
 	}
 
+	if oldState == connectivity.Connecting && newState.ConnectivityState == connectivity.TransientFailure {
+		pickFirstConnectionAttemptsFailed.Record(b.metricsRecorder, 1, b.target)
```

Comment:
Does this work if the passed in metrics recorder is `nil`? I'm guessing not, since I see changes in tests setting this to nop metrics recorder. Can we instead handle that in `Build`?

Reply:
> the top level channel/server creates a metrics recorder list and sends it down, so it should never be nil

If that is the case, maybe just a comment somewhere up top (maybe along with the field that it is guaranteed to be non-nil) should be good enough

Reply:
Done. Thanks.